Frontend/src/auth/context/firebase/auth-provider.tsx

```diff
@@ -0,0 +1,214 @@
+      user: action.payload.user,
+    };
+  }
+  return state;
+};
+
+// ----------------------------------------------------------------------
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export function AuthProvider({ children }: Props) {
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  const initialize = useCallback(() => {
+    // signOut(AUTH);
+    try {
+      onAuthStateChanged(AUTH, async (user) => {
+        if (user) {
+          if (user.emailVerified) {
+            const accessToken = await user.getIdToken();
+            const isRegistered = await isRegister(accessToken);
+            dispatch({
+              type: Types.INITIAL,
+              payload: {
+                user: {
+                  ...user,
+                  isRegistered,
+                  id: user.uid,
+                  role: 'student',
+                },
+              },
+            });
+          } else {
+            dispatch({
+              type: Types.INITIAL,
+              payload: {
+                user: null,
+              },
+            });
+          }
+        } else {
+          dispatch({
+            type: Types.INITIAL,
+            payload: {
+              user: null,
+            },
+          });
+        }
+      });
+    } catch (error) {
+      console.error(error);
+      dispatch({
+        type: Types.INITIAL,
+        payload: {
+          user: null,
+        },
+      });
+    }
+  }, []);
+
+  useEffect(() => {
+    initialize();
+  }, [initialize]);
+
+  // LOGIN
+  const loginWithGoogle = useCallback(async () => {
+    const provider = new GoogleAuthProvider();
+
+    await signInWithPopup(AUTH, provider);
+  }, []);
+
+  const loginWithFacebook = useCallback(async () => {
+    const provider = new FacebookAuthProvider();
+
+    await signInWithPopup(AUTH, provider);
+  }, []);
+
+  const loginWithMicrosoft = useCallback(async () => {
+    const provider = new OAuthProvider('microsoft.com');
+
+    await signInWithPopup(AUTH, provider);
+  }, []);
+
+  // REGISTER
+  const register = useCallback(
+    async (firstName: string, lastName: string): Promise<void> => {
+      const data = {
+        firstName,
+        lastName,
+      };
+
+      if (!data.firstName || !data.lastName) {
+        console.error('first name and last name are required!!!');
+      } else {
+        const accessToken = state.user?.accessToken;
+        console.table({ 'access token': accessToken, ...data });
+        // try {
+        //   const res = await axios.post(endpoints.auth.register, data, {
+        //     headers: {
+        //       Authorization: `Bearer ${accessToken}`,
+        //       'Content-Type': 'application/json',
+        //     },
+        //   });
+        //   const user = await res.data;
+        //   dispatch({
+        //     type: Types.REGISTER,
+        //     payload: {
+        //       user: {
+        //         ...user,
+        //       },
+        //     },
+        //   });
```

Comment:
lots of comments. I don't like comments in the code

Reply:
### IN THE DESCRIPTION ABOVE:

> I have added the logic to use the continue registration API in the auth-provider.tsx file. However, I have commented it out for now.

- Because the error we talked about existed in the backend for creating an account, so I added the required logic and commented it out so that the interface would work temporarily while Ammar fixed the error.
I apologize for these comments.